### ISSUE TITLE
D405 & D407 false-positive parameters recognition

### ISSUE THREAD

**Issue by Liimpo** [PyCQA/pydocstyle#481]
I am currently working in a project where we use type annotations [PEP 484](https://www.python.org/dev/peps/pep-0484/) and [pdoc3](https://pypi.org/project/pdoc3/) to create our documentations. We check our code with this awesome checker.  
The error I am getting is from the following code.  
```Python
def foo(a: str, b: bool, parameters: Dict[str, Any]) -> str:
    """I am a cool function.

    Parameters
    ----------
    a
        Test 1.
    b
        Test 2.
    parameters
        Test 3.
    Returns
    -------
    str
        Something neat.
```

The error seems to arise when parameters is used under the Parameters section.  It only occurs when using type annotations following PEP 484. When I add the type after parameters as. `parameters: Dict[str, any]` in the body I get no errors.
